### ISSUE TITLE
Negative loss more clearly explanation

### ISSUE THREAD

**Issue by xiaogangzhu** [cpmpercussion/keras-mdn-layer#42]
Hi,

I am trying to understand why the mdn loss here may be negative and hope you can help me!

I learned that the mdn loss function is defined as the negative log loss of PDF, the PDF is always bigger than 0 and smaller or equal to 1. Thus the negative log loss of PDF will always be positive. Why the negative loss is normal here.

I also check the get_mixture_loss_func and still cannot fully understand the negative loss logic, is the tfd.Mixture matters here?

Hope you can help me!

**Comment by cpmpercussion** [comment]
> the PDF is always bigger than 0 and smaller or equal to 1

This is not a correct assumption for the PDF of a continuous probability distribution.

These functions can end up higher than 1 at some points (leading to less that 0 loss), as long as the area under the curve is equal to one.

You can try this if you like with the PDF for a Gaussian distribution with a very small standard deviation. As the sd approaches zero, the PDF goes to infinity at the mean. For example, try the following standard python code:
```
## Normal Distribution PDF can be > 1
from scipy.stats import norm

norm.pdf(0, loc=0, scale=0.1) # returns 3.989422804014327
```

It’s bad if the PDF in the loss function goes to infinity, so we add a small number to the standard deviations in our calculations to make sure that doesn’t happen.
